src/js/application.js

```diff
@@ -0,0 +1,317 @@
+        this.showTemperatureForecast(this.currentWeather);  
+        this.showNextDaysOfWeek(this.currentWeather);  
+        this.showWeatherIcons(this.currentWeather); 
+    }
+
+    async setLanguage(language, updateLocation = true) {
+        this.language = language;
+        const selectLanguageElement = document.querySelector(`[value=${this.language}]`);
+        selectLanguageElement.setAttribute('selected', 'selected');
+
+        this.translateContent();
+        if (updateLocation) {
+            const location = new Location();
+            await location.defineLocationByName(this.currentLocation.cityName, this.language);
+            this.currentLocation = location.getLocationData(); 
+            
+            const time = new Time(this.currentLocation.latitude, this.currentLocation.longitude, this.currentLocation.language);
+            await time.loadLocationTimezone();
+            const timeZone = time.getTimeZone();
+            await this.setLocation(this.currentLocation, timeZone);                        
+        }
+    }
+    
+    changeLanguage(e) {
+        const language = e.target.value;
+        localStorage.setItem(localStorageItems.language, language);
+        this.setLanguage(language); 
+    }
+    
+    translateContent() {
+        const stringsToBeResolved = document.querySelectorAll('[data-content]');
+
+        stringsToBeResolved.forEach(el => {
+            el.textContent = translation[el.attributes['data-content'].value][this.language];
+        });
+    }
+
+    markAsNotActive(el) {
+        el.classList.remove(classes.ACTIVE);
+        el.classList.add(classes.NOT_ACTIVE);
+    }
+    
+    markAsActive(el) {
+        el.classList.remove(classes.NOT_ACTIVE);
+        el.classList.add(classes.ACTIVE);
+    }
+
+    initActiveDegreeButton() {
+        const fahrenheitTemperatureBtn = document.querySelector(selectors.FAHRENHEIT_TEMPERATURE_BTN);
+        const celsiusTemperatureBtn = document.querySelector(selectors.CELSIUS_TEMPERATURE_BTN);
+
+        if (this.degreeUnit === degreeUnits.fahrenheit) {
+            this.markAsNotActive(celsiusTemperatureBtn);
+            this.markAsActive(fahrenheitTemperatureBtn);
+        } else {
+            this.markAsNotActive(fahrenheitTemperatureBtn);
+            this.markAsActive(celsiusTemperatureBtn);
+        }
+    }
+
+    determineClickedButton(e) {
+        const temperatureButtons = document.querySelectorAll(selectors.TEMPERATURE_BTN);
+        
+        if (e.target.className.includes(classes.CELSIUS) ||
+            e.target.className.includes(classes.FAHRENHEIT)) {                
+                temperatureButtons.forEach((el) => {
+                    this.markAsNotActive(el);
+
+                    if (el.className.includes(e.target.className)) {
+                        this.markAsActive(el);
+                    }                  
+                });
+                this.degreeUnit = e.target.className;
```

Comment:
you use  e.target.className all the time, please add variable for it

additionally if condition can be written in one line (line 301-302)

Reply:
fixed